### ISSUE TITLE
Timezone changes result in duplicate offset data

### ISSUE THREAD

**Issue by scottleibrand** [nightscout/android-uploader#126]
To reproduce what we saw flying to Hawaii and back, start with the Dexcom receiver and Android uploader phone both set to the correct time in your home timezone.  Then, switch the uploader phone's timezone (making sure the time remains correct for the new timezone) and update the Dexcom's time accordingly.  Then, do a gap sync.

When you do that, you'll see that the data displayed in Nightscout is all duplicated, with an offset of however many hours are between the two timezones.

The desired behavior would be that the android-uploader matches up the Dexcom receiver's time with the local time on the phone, and if they match, inserts new data using the local timezone.  It also needs to make sure that data uploaded by gap sync after a Dexcom receiver time(zone) change is not an offset duplicate of already-uploaded data.  (I'm not sure whether/how the internal timestamps change when you change the receiver's display time, but I believe at least one of the timestamps is unaffected, so we should be able to use that to verify no offset duplication is occurring.)

If this isn't enough detail to reproduce the problem or identify the solution, LMK and I'll dig into what exactly is happening in the uploaded data, and figure out exactly what it should be doing instead.

**Comment by ktind** [comment]
This is a known issue. See [here](https://github.com/nightscout/android-uploader/wiki/Time-and-the-Dexcom) for more details.

We are experimenting with [Solution 2](https://github.com/nightscout/android-uploader/wiki/Time-and-the-Dexcom#solution-2---use-the-system-time) in [wip/mqtt/init](https://github.com/nightscout/cgm-remote-monitor/tree/wip/mqtt/init) using the uploader branch [kl.mqttui](https://github.com/nightscout/android-uploader/tree/76-kl.mqttui)

Once I have confirmation from a broader user base that the issue is resolved I will work on updating REST and Mongo to use this algorithm.

**Comment by cjbeck** [comment]
I just uploaded some logs from my motoG, which seems to have the 1-2 hour offset issue. My android tablet (N9) works perfectly fine (also running the latest code). I'm not sure what to do in order to get the MotoG back to a working state. It's strange, because it was working fine yesterday, but I think I messed with the values under preferences for Sensor Data and Calibration Data, which I think could be the cause of this.
